ramanujan/cmf/cmf.py

```diff
@@ -28,9 +29,22 @@ def __init__(self, Mx: Matrix, My: Matrix):
         if simplify(Mxy - Myx) != Matrix([[0, 0], [0, 0]]):
             raise ValueError("The given Mx and My matrices are not conserving!")
 
+        self.initial_loc = initial_loc
+        if potential_cache_file is not None:
+            self.potential_cache = self.load_potential_from_file(potential_cache_file)
+        else:
+            self.reset_potential_cache(maximal_cache_dims, initial_mat)
+
     def __repr__(self):
         return f"CMF({self.Mx}, {self.My})"
 
+    def reset_potential_cache(self, maximal_cache_dims, initial_mat):
```

Comment:
If you create this new "cache" object, you can use it here as well. This "reset" method, should probably be a method of this new object